### PR TITLE
Port breakout Show Command

### PR DESCRIPTION
**Pull request by samaity**
Signed-off-by: Sangita Maity <sangitamaity0211@gmail.com>

**- What I did**
Implemented `show interface breakout` sub-command to show the port capability and the current breakout mode.

Available commands are mentioned below.
```
show interfaces breakout --help
show interfaces breakout current-mode
```

**- How I did it**
using `platform.json`, `hwsku.json` and config-db `BREAKOUT_CFG` table.

**- How to verify it**
```
admin@lnos-x1-a-fab01:~$ show interfaces breakout --help
Usage: show interfaces breakout [OPTIONS] COMMAND [ARGS]...

  Show interface breakout

Options:
  -?, -h, --help  Show this message and exit.

Commands:
  current-mode  Show interface breakout current-mode
```
```
admin@lnos-x1-a-fab01:~$ show interfaces breakout
{
    "Ethernet0": {
        "index": "1,1,1,1",
        "default_brkout_mode": "4x25G[10G]",
        "lanes": "65,66,67,68",
        "brkout_mode": "4x25G[10G]",
        "breakout_modes": "1x100G[40G],2x50G,4x25G[10G]",
        "alias_at_lanes": "Eth1/1, Eth1/2, Eth1/3, Eth1/4"
    },
    "Ethernet4": {
        "index": "2,2,2,2",
        "default_brkout_mode": "4x25G[10G]",
        "lanes": "69,70,71,72",
        "brkout_mode": "4x25G[10G]",
        "breakout_modes": "1x100G[40G],2x50G,4x25G[10G]",
        "alias_at_lanes": "Eth2/1, Eth2/2, Eth2/3, Eth2/4"
    },
    "Ethernet8": {
        "index": "3,3,3,3",
        "default_brkout_mode": "4x25G[10G]",
        "lanes": "73,74,75,76",
        "brkout_mode": "4x25G[10G]",
        "breakout_modes": "1x100G[40G],2x50G,4x25G[10G]",
        "alias_at_lanes": "Eth3/1, Eth3/2, Eth3/3, Eth3/4"
    },
    "Ethernet12": {
        "index": "4,4,4,4",
        "default_brkout_mode": "4x25G[10G]",
        "lanes": "77,78,79,80",
        "brkout_mode": "4x25G[10G]",
        "breakout_modes": "1x100G[40G],2x50G,4x25G[10G]",
        "alias_at_lanes": "Eth4/1, Eth4/2, Eth4/3, Eth4/4"
    },
    "Ethernet16": {
        "index": "5,5,5,5",
        "default_brkout_mode": "4x25G[10G]",
        "lanes": "33,34,35,36",
        "brkout_mode": "4x25G[10G]",
        "breakout_modes": "1x100G[40G],2x50G,4x25G[10G]",
        "alias_at_lanes": "Eth5/1, Eth5/2, Eth5/3, Eth5/4"
    },
    "Ethernet20": {
        "index": "6,6,6,6",
        "default_brkout_mode": "4x25G[10G]",
        "lanes": "37,38,39,40",
        "brkout_mode": "4x25G[10G]",
        "breakout_modes": "1x100G[40G],2x50G,4x25G[10G]",
        "alias_at_lanes": "Eth6/1, Eth6/2, Eth6/3, Eth6/4"
    },
    "Ethernet24": {
        "index": "7,7,7,7",
        "default_brkout_mode": "4x25G[10G]",
        "lanes": "41,42,43,44",
        "brkout_mode": "4x25G[10G]",
        "breakout_modes": "1x100G[40G],2x50G,4x25G[10G]",
        "alias_at_lanes": "Eth7/1, Eth7/2, Eth7/3, Eth7/4"
    },
    "Ethernet28": {
        "index": "8,8,8,8",
        "default_brkout_mode": "4x25G[10G]",
        "lanes": "45,46,47,48",
        "brkout_mode": "4x25G[10G]",
        "breakout_modes": "1x100G[40G],2x50G,4x25G[10G]",
        "alias_at_lanes": "Eth8/1, Eth8/2, Eth8/3, Eth8/4"
    },
    "Ethernet32": {
        "index": "9,9,9,9",
        "default_brkout_mode": "4x25G[10G]",
        "lanes": "49,50,51,52",
        "brkout_mode": "4x25G[10G]",
        "breakout_modes": "1x100G[40G],2x50G,4x25G[10G]",
        "alias_at_lanes": "Eth9/1, Eth9/2, Eth9/3, Eth9/4"
    },
    "Ethernet36": {
        "index": "10,10,10,10",
        "default_brkout_mode": "4x25G[10G]",
        "lanes": "53,54,55,56",
        "brkout_mode": "4x25G[10G]",
        "breakout_modes": "1x100G[40G],2x50G,4x25G[10G]",
        "alias_at_lanes": "Eth10/1, Eth10/2, Eth10/3, Eth10/4"
    },
    "Ethernet40": {
        "index": "11,11,11,11",
        "default_brkout_mode": "4x25G[10G]",
        "lanes": "57,58,59,60",
        "brkout_mode": "4x25G[10G]",
        "breakout_modes": "1x100G[40G],2x50G,4x25G[10G]",
        "alias_at_lanes": "Eth11/1, Eth11/2, Eth11/3, Eth11/4"
    },
    "Ethernet44": {
        "index": "12,12,12,12",
        "default_brkout_mode": "4x25G[10G]",
        "lanes": "61,62,63,64",
        "brkout_mode": "4x25G[10G]",
        "breakout_modes": "1x100G[40G],2x50G,4x25G[10G]",
        "alias_at_lanes": "Eth12/1, Eth12/2, Eth12/3, Eth12/4"
    },
    "Ethernet48": {
        "index": "13,13,13,13",
        "default_brkout_mode": "4x25G[10G]",
        "lanes": "81,82,83,84",
        "brkout_mode": "4x25G[10G]",
        "breakout_modes": "1x100G[40G],2x50G,4x25G[10G]",
        "alias_at_lanes": "Eth13/1, Eth13/2, Eth13/3, Eth13/4"
    },
    "Ethernet52": {
        "index": "14,14,14,14",
        "default_brkout_mode": "4x25G[10G]",
        "lanes": "85,86,87,88",
        "brkout_mode": "4x25G[10G]",
        "breakout_modes": "1x100G[40G],2x50G,4x25G[10G]",
        "alias_at_lanes": "Eth14/1, Eth14/2, Eth14/3, Eth14/4"
    },
    "Ethernet56": {
        "index": "15,15,15,15",
        "default_brkout_mode": "4x25G[10G]",
        "lanes": "89,90,91,92",
        "brkout_mode": "4x25G[10G]",
        "breakout_modes": "1x100G[40G],2x50G,4x25G[10G]",
        "alias_at_lanes": "Eth15/1, Eth15/2, Eth15/3, Eth15/4"
    },
    "Ethernet60": {
        "index": "16,16,16,16",
        "default_brkout_mode": "4x25G[10G]",
        "lanes": "93,94,95,96",
        "brkout_mode": "4x25G[10G]",
        "breakout_modes": "1x100G[40G],2x50G,4x25G[10G]",
        "alias_at_lanes": "Eth16/1, Eth16/2, Eth16/3, Eth16/4"
    },
    "Ethernet64": {
        "index": "17,17,17,17",
        "default_brkout_mode": "4x25G[10G]",
        "lanes": "97,98,99,100",
        "brkout_mode": "4x25G[10G]",
        "breakout_modes": "1x100G[40G],2x50G,4x25G[10G]",
        "alias_at_lanes": "Eth17/1, Eth17/2, Eth17/3, Eth17/4"
    },
    "Ethernet68": {
        "index": "18,18,18,18",
        "default_brkout_mode": "4x25G[10G]",
        "lanes": "101,102,103,104",
        "brkout_mode": "4x25G[10G]",
        "breakout_modes": "1x100G[40G],2x50G,4x25G[10G]",
        "alias_at_lanes": "Eth18/1, Eth18/2, Eth18/3, Eth18/4"
    },
    "Ethernet72": {
        "index": "19,19,19,19",
        "default_brkout_mode": "4x25G[10G]",
        "lanes": "105,106,107,108",
        "brkout_mode": "4x25G[10G]",
        "breakout_modes": "1x100G[40G],2x50G,4x25G[10G]",
        "alias_at_lanes": "Eth19/1, Eth19/2, Eth19/3, Eth19/4"
    },
    "Ethernet76": {
        "index": "20,20,20,20",
        "default_brkout_mode": "4x25G[10G]",
        "lanes": "109,110,111,112",
        "brkout_mode": "4x25G[10G]",
        "breakout_modes": "1x100G[40G],2x50G,4x25G[10G]",
        "alias_at_lanes": "Eth20/1, Eth20/2, Eth20/3, Eth20/4"
    },
    "Ethernet80": {
        "index": "21,21,21,21",
        "default_brkout_mode": "4x25G[10G]",
        "lanes": "1,2,3,4",
        "brkout_mode": "4x25G[10G]",
        "breakout_modes": "1x100G[40G],2x50G,4x25G[10G]",
        "alias_at_lanes": "Eth21/1, Eth21/2, Eth21/3, Eth21/4"
    },
    "Ethernet84": {
        "index": "22,22,22,22",
        "default_brkout_mode": "4x25G[10G]",
        "lanes": "5,6,7,8",
        "brkout_mode": "4x25G[10G]",
        "breakout_modes": "1x100G[40G],2x50G,4x25G[10G]",
        "alias_at_lanes": "Eth22/1, Eth22/2, Eth22/3, Eth22/4"
    },
    "Ethernet88": {
        "index": "23,23,23,23",
        "default_brkout_mode": "4x25G[10G]",
        "lanes": "9,10,11,12",
        "brkout_mode": "4x25G[10G]",
        "breakout_modes": "1x100G[40G],2x50G,4x25G[10G]",
        "alias_at_lanes": "Eth23/1, Eth23/2, Eth23/3, Eth23/4"
    },
    "Ethernet92": {
        "index": "24,24,24,24",
        "default_brkout_mode": "4x25G[10G]",
        "lanes": "13,14,15,16",
        "brkout_mode": "4x25G[10G]",
        "breakout_modes": "1x100G[40G],2x50G,4x25G[10G]",
        "alias_at_lanes": "Eth24/1, Eth24/2, Eth24/3, Eth24/4"
    },
    "Ethernet96": {
        "index": "25,25,25,25",
        "default_brkout_mode": "2x50G",
        "lanes": "17,18,19,20",
        "brkout_mode": "2x50G",
        "breakout_modes": "1x100G[40G],2x50G,4x25G[10G]",
        "alias_at_lanes": "Eth25/1, Eth25/2, Eth25/3, Eth25/4"
    },
    "Ethernet100": {
        "index": "26,26,26,26",
        "default_brkout_mode": "2x50G",
        "lanes": "21,22,23,24",
        "brkout_mode": "2x50G",
        "breakout_modes": "1x100G[40G],2x50G,4x25G[10G]",
        "alias_at_lanes": "Eth26/1, Eth26/2, Eth26/3, Eth26/4"
    },
    "Ethernet104": {
        "index": "27,27,27,27",
        "default_brkout_mode": "2x50G",
        "lanes": "25,26,27,28",
        "brkout_mode": "2x50G",
        "breakout_modes": "1x100G[40G],2x50G,4x25G[10G]",
        "alias_at_lanes": "Eth27/1, Eth27/2, Eth27/3, Eth27/4"
    },
    "Ethernet108": {
        "index": "28,28,28,28",
        "default_brkout_mode": "2x50G",
        "lanes": "29,30,31,32",
        "brkout_mode": "2x50G",
        "breakout_modes": "1x100G[40G],2x50G,4x25G[10G]",
        "alias_at_lanes": "Eth28/1, Eth28/2, Eth28/3, Eth28/4"
    },
    "Ethernet112": {
        "index": "29,29,29,29",
        "default_brkout_mode": "2x50G",
        "lanes": "113,114,115,116",
        "brkout_mode": "2x50G",
        "breakout_modes": "1x100G[40G],2x50G,4x25G[10G]",
        "alias_at_lanes": "Eth29/1, Eth29/2, Eth29/3, Eth29/4"
    },
    "Ethernet116": {
        "index": "30,30,30,30",
        "default_brkout_mode": "2x50G",
        "lanes": "117,118,119,120",
        "brkout_mode": "2x50G",
        "breakout_modes": "1x100G[40G],2x50G,4x25G[10G]",
        "alias_at_lanes": "Eth30/1, Eth30/2, Eth30/3, Eth30/4"
    },
    "Ethernet120": {
        "index": "31,31,31,31",
        "default_brkout_mode": "2x50G",
        "lanes": "121,122,123,124",
        "brkout_mode": "2x50G",
        "breakout_modes": "1x100G[40G],2x50G,4x25G[10G]",
        "alias_at_lanes": "Eth31/1, Eth31/2, Eth31/3, Eth31/4"
    },
    "Ethernet124": {
        "index": "32,32,32,32",
        "default_brkout_mode": "2x50G",
        "lanes": "125,126,127,128",
        "brkout_mode": "2x50G",
        "breakout_modes": "1x100G[40G],2x50G,4x25G[10G]",
        "alias_at_lanes": "Eth32/1, Eth32/2, Eth32/3, Eth32/4"
    }
}
admin@lnos-x1-a-fab01:~$
```

```

admin@lnos-x1-a-fab01:~$ show interfaces breakout current-mode Ethernet0
+-------------+-------------------------+
| Interface   | Current Breakout Mode   |
+=============+=========================+
| Ethernet0   | 4x25G[10G]              |
+-------------+-------------------------+
```
```
admin@lnos-x1-a-fab01:~$ show interfaces breakout current-mode
+-------------+-------------------------+
| Interface   | Current Breakout Mode   |
+=============+=========================+
| Ethernet0   | 4x25G[10G]              |
+-------------+-------------------------+
| Ethernet4   | 4x25G[10G]              |
+-------------+-------------------------+
| Ethernet8   | 4x25G[10G]              |
+-------------+-------------------------+
| Ethernet12  | 4x25G[10G]              |
+-------------+-------------------------+
| Ethernet16  | 4x25G[10G]              |
+-------------+-------------------------+
| Ethernet20  | 4x25G[10G]              |
+-------------+-------------------------+
| Ethernet24  | 4x25G[10G]              |
+-------------+-------------------------+
| Ethernet28  | 4x25G[10G]              |
+-------------+-------------------------+
| Ethernet32  | 4x25G[10G]              |
+-------------+-------------------------+
| Ethernet36  | 4x25G[10G]              |
+-------------+-------------------------+
| Ethernet40  | 4x25G[10G]              |
+-------------+-------------------------+
| Ethernet44  | 4x25G[10G]              |
+-------------+-------------------------+
| Ethernet48  | 4x25G[10G]              |
+-------------+-------------------------+
| Ethernet52  | 4x25G[10G]              |
+-------------+-------------------------+
| Ethernet56  | 4x25G[10G]              |
+-------------+-------------------------+
| Ethernet60  | 4x25G[10G]              |
+-------------+-------------------------+
| Ethernet64  | 4x25G[10G]              |
+-------------+-------------------------+
| Ethernet68  | 4x25G[10G]              |
+-------------+-------------------------+
| Ethernet72  | 4x25G[10G]              |
+-------------+-------------------------+
| Ethernet76  | 4x25G[10G]              |
+-------------+-------------------------+
| Ethernet80  | 4x25G[10G]              |
+-------------+-------------------------+
| Ethernet84  | 4x25G[10G]              |
+-------------+-------------------------+
| Ethernet88  | 4x25G[10G]              |
+-------------+-------------------------+
| Ethernet92  | 4x25G[10G]              |
+-------------+-------------------------+
| Ethernet96  | 2x50G                   |
+-------------+-------------------------+
| Ethernet100 | 2x50G                   |
+-------------+-------------------------+
| Ethernet104 | 2x50G                   |
+-------------+-------------------------+
| Ethernet108 | 2x50G                   |
+-------------+-------------------------+
| Ethernet112 | 2x50G                   |
+-------------+-------------------------+
| Ethernet116 | 2x50G                   |
+-------------+-------------------------+
| Ethernet120 | 2x50G                   |
+-------------+-------------------------+
| Ethernet124 | 2x50G                   |
+-------------+-------------------------+
```

gracefully handled all the errors like where BREAKOUT_CFG is not present in configDB, platform.json/hwsku.json are not available etc.
```

admin@lnos-x1-a-fab01:~$ show interfaces breakout
[Errno 2] No such file or directory: '/usr/share/sonic/device/x86_64-cel_seastone-r0/platform.json'
Aborted!
```

### CODE REVIEW

File: show/main.py [hunk 1]
```diff
@@ -796,6 +810,68 @@ def alias(interfacename):
...
+    # Show current Breakout Mode of user prompted interface
+    if interface is not None:
+        body.append([interface, str(curBrkout_tbl[interface]['brkout_mode'])])
+        click.echo(tabulate(body, header, tablefmt="grid"))
+        return
+
+    # Show current Breakout Mode for all interfaces
+    for name in curBrkout_tbl.keys():
+        body.append([name, str(curBrkout_tbl[name]['brkout_mode'])])
+    click.echo(tabulate(body, header, tablefmt="grid"))
```
Comment (zhenggen-xu) [inline show/main.py:884]: Please nature sort the interface names here.

File: show/main.py [hunk 2]
```diff
@@ -796,6 +810,68 @@ def alias(interfacename):
...
+@interfaces.group(invoke_without_command=True)
+@click.pass_context
+def breakout(ctx):
+    """Show interface breakout"""
+
+    config_db = ConfigDBConnector()
+    config_db.connect()
+    ctx.obj = {'db': config_db}
+
+    curBrkout_tbl = config_db.get_table('BREAKOUT_CFG')
```
Comment (zhenggen-xu) [inline show/main.py]: We need gracefully handle the cases where BREAKOUT_CFG is not in configDB,  platform.json/hwsku.json are not available cases.

**Review by zhenggen-xu — CHANGES_REQUESTED** [review]

File: show/main.py (hunk 1 above)
Comment (samaity) [reply show/main.py:884]: done

File: show/main.py (hunk 2 above)
Comment (samaity) [reply show/main.py]: done

**Review by zhenggen-xu — APPROVED** [review]

**Pull request merged**